group/assignment5/AlphabetFromDictionary.cpp

```diff
@@ -0,0 +1,94 @@
+#include "AlphabetFromDictionary.hpp"
+
+void AlphabetFromDictionary::setDictionary(vector<string> dictionary) {
+    this->dictionary = dictionary;
+}
+
+void AlphabetFromDictionary::initializeClassVars() {
+    vector<char> alphabetCharacters = findAlphabetCharacters();
+    this->charPriorityGraph.clear();
+    this->numOfPriorCharacters.clear();
+    for (auto c : alphabetCharacters) {
+        this->charPriorityGraph[c] = set<char>();
+        this->numOfPriorCharacters[c] = 0;
+    }
+}
+
+bool AlphabetFromDictionary::findAlphabet(vector<char> &alphabet) {

```

Comment:
Do we want to let user to call this function multiple times? If dictionary is the same, graphCreation needlessly will be called. One option is to have boolean flag which will tell us do we need to execute this function again (if dictionary is changed we can reset that flag) and in the other option we can call this function from constructor and from setDictionary function and make it private, so the user won't be able to call it by himself. We can provide only getAlfabet() function. What do you think? :)

Reply:
Yes, definitely we can let user to change the dictionary, but I guess we don't want to let him call this function multiple times over the same dictionary? One option was to add flag and in the other one we will call that function only from constructor and from setDictionary methods.

Reply:
I agree about not calling findAlphabet() multiple times.
I think I actually prefer the second option you suggested - calling findAlphabet() from the constructor and from setDictionary()

Reply:
Me too :) Lets see what other girls think

Reply:
I think for this assignment calling findAphabet only once for the dictionary is just ok, so I agree with you :)

Reply:
I don't agree, first if dictionary is the same and flag tells that, where is the alphabet? it is not saved in the class so it should find it again anyways. Even if you pass it to setDictionary or constructor you are not saving it in the class so how can you return it to user whenever she needs it? which will be at setDictionary? How can you obtain alphabet from class when you are not changing dictionary?

Reply:
We will save the current alphabet in the class and provide a getAlphabet() function for the user